core/state/statedb.go

```diff
@@ -802,28 +802,51 @@ func (s *StateDB) clearJournalAndRefund() {
 	s.validRevisions = s.validRevisions[:0] // Snapshots can be created without journal entires
 }
 
+// ModifiedAccount is an Account and it's Storage trie that has changed in the current block.
+type ModifiedAccount struct {
+	Account
+	Storage
+}
+
+// StateChanges are a map between an Account's address to it's ModifiedAccount.
+type StateChanges map[common.Address]ModifiedAccount
+
 // Commit writes the state to the underlying in-memory trie database.
-func (s *StateDB) Commit(deleteEmptyObjects bool) (common.Hash, error) {
+func (s *StateDB) Commit(deleteEmptyObjects bool) (common.Hash, StateChanges, error) {
```

Comment:
These are changes in a very "core" place, and I'd estimate that they're very heavy changes -- this would probably slow down block processing and/or bump memory consumption quite a lot

Reply:
Thanks for the feedback @holiman - also mentioned this in response to @karalabe below, but do you think that adding a CLI flag for users to be able to explicitly opt-in to this additional data being returned from `Commit` would be helpful?